### PR TITLE
KIP-16331: Remove EOS-v1 from StreamsConfigUtils

### PR DESCRIPTION
**Pull request by mjsax**
Call for review @bbejeck 

This should be the last PR for EOSv1 removal.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java [hunk 1]
```diff
@@ -274,11 +274,7 @@ public void pauseTopology(final String topologyName) {
      * @return A boolean indicating if the topology is paused.
      */
     public boolean isPaused(final String topologyName) {
-        if (topologyName == null) {
-            return pausedTopologies.contains(UNNAMED_TOPOLOGY);
-        } else {
-            return pausedTopologies.contains(topologyName);
-        }
+        return pausedTopologies.contains(getTopologyNameOrElseUnnamed(topologyName));
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java:277]: Side cleanup

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java [hunk 2]
```diff
@@ -463,7 +459,7 @@ Pattern sourceTopicPattern() {
 
         applyToEachBuilder(b -> {
             final String patternString = b.sourceTopicPatternString();
-            if (patternString.length() > 0) {
+            if (!patternString.isEmpty()) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java:462]: Side cleanup

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java [hunk 3]
```diff
@@ -629,11 +625,7 @@ public Collection<NamedTopology> allNamedTopologies() {
      *         else returns {@code null} if {@code topologyName} is non-null but no such NamedTopology exists
      */
     public InternalTopologyBuilder lookupBuilderForNamedTopology(final String topologyName) {
-        if (topologyName == null) {
-            return builders.get(UNNAMED_TOPOLOGY);
-        } else {
-            return builders.get(topologyName);
-        }
+        return builders.get(getTopologyNameOrElseUnnamed(topologyName));
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java:628]: Side cleanup

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java [hunk 4]
```diff
@@ -135,8 +134,8 @@ void initTransaction() {
     }
 
     public void resetProducer(final Producer<byte[], byte[]> producer) {
-        if (processingMode != EXACTLY_ONCE_V2) {
-            throw new IllegalStateException("Expected eos-v2 to be enabled, but the processing mode was " + processingMode);
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java:139]: Any way to put back the misconfigured `processingMode`? It's a minor point but could save a user from having to look up the current configs.  I'll leave it up to you.

**Review by bbejeck — APPROVED** [review]
LGTM only one minor comment

**Pull request merged**